framework/src/headers/ip/v6/icmp/router_advertisement.rs

```diff
@@ -0,0 +1,286 @@
+
+      Source link-layer address
+                     The link-layer address of the interface from which
+                     the Router Advertisement is sent.  Only used on
+                     link layers that have addresses.  A router MAY omit
+                     this option in order to enable inbound load sharing
+                     across multiple link-layer addresses.
+
+      MTU            SHOULD be sent on links that have a variable MTU
+                     (as specified in the document that describes how to
+                     run IP over the particular link type).  MAY be sent
+                     on other links.
+
+      Prefix Information
+                     These options specify the prefixes that are on-link
+                     and/or are used for stateless address
+                     autoconfiguration.  A router SHOULD include all its
+                     on-link prefixes (except the link-local prefix) so
+                     that multihomed hosts have complete prefix
+                     information about on-link destinations for the
+                     links to which they attach.  If complete
+                     information is lacking, a host with multiple
+                     interfaces may not be able to choose the correct
+                     outgoing interface when sending traffic to its
+                     neighbors.
+      Future versions of this protocol may define new option types.
+      Receivers MUST silently ignore any options they do not recognize
+      and continue processing the message.
+
+*/
+
+const MANAGED_CFG_ADDR_POS: u8 = 0;
+const OTHER_CFG_POS: u8 = 1;
+
+#[derive(Debug)]
+#[repr(C, packed)]
+pub struct Icmpv6RouterAdvertisement<T>
+where
+    T: Ipv6VarHeader,
+{
+    icmp: Icmpv6Header<T>,
+    current_hop_limit: u8,
+    reserved_flags: u8,
+    router_lifetime: u16,
+    reachable_time: u32,
+    retrans_timer: u32,
+    options: u8,
+    _parent: PhantomData<T>,
+}
+
+impl<T> Default for Icmpv6RouterAdvertisement<T>
+where
+    T: Ipv6VarHeader,
+{
+    fn default() -> Icmpv6RouterAdvertisement<T> {
+        Icmpv6RouterAdvertisement {
+            icmp: Icmpv6Header {
+                msg_type: IcmpMessageType::RouterAdvertisement as u8,
+                code: 0,
+                checksum: 0,
+                ..Default::default()
+            },
+            current_hop_limit: 0,
+            reserved_flags: 0,
+            router_lifetime: 0,
+            reachable_time: 0,
+            retrans_timer: 0,
+            options: 0,
+            _parent: PhantomData,
+        }
+    }
+}
+
+impl<T> fmt::Display for Icmpv6RouterAdvertisement<T>
+where
+    T: Ipv6VarHeader,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "msg_type: {} code: {} checksum: {}, current_hop_limit {}, reserved_flags {}, router_lifetime {}, reachable_time {}, retrans_timers {}",
+            self.msg_type().unwrap(),
```

Comment:
outside scope of this pr but I don't like the way we are modeling type enumerations.